packages/block-editor/src/components/block-list/block.js

```diff
@@ -590,9 +595,14 @@ function BlockListBlockProvider( props ) {
 				index: getBlockIndex( clientId ),
 				blockApiVersion: blockType?.apiVersion || 1,
 				blockTitle: match?.title || blockType?.title,
-				isSubtreeDisabled: isBlockSubtreeDisabled( clientId ),
+				isSubtreeDisabled:
+					blockEditingMode === 'disabled' &&
+					isBlockSubtreeDisabled( clientId ),
 				isOutlineEnabled: outlineMode,
-				hasOverlay: __unstableHasActiveBlockOverlayActive( clientId ),
+				hasOverlay: __unstableHasActiveBlockOverlayActive(
+					clientId,
+					blockEditingMode
+				),
```

Comment:
This is an unstable selector, so it seems ok to add optional parameters for now.